posthog/models/kafka_partition_stats/sql.py

```diff
@@ -83,14 +94,69 @@
 """
 )
 
-CREATE_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = CREATE_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 CREATE_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = CREATE_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
 
-DROP_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = DROP_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 DROP_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = DROP_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
+
+# V2
+
+
+class PartitionStatsV2Table:
+    table_name: str = "events_plugin_ingestion_partition_statistics_v2"
+
+    def get_create_table_sql(self) -> str:
+        engine = ReplacingMergeTree(self.table_name, ver="timestamp")
+        return f"""
+            CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+                topic LowCardinality(String),
+                partition UInt64,
+                offset UInt64,
+                token String,
+                distinct_id String,
+                ip String,
+                event String,
+                data_length UInt64,
+                timestamp DateTime
```

Comment:
Timestamp probably would compress relatively well with either delta or Gorilla encoding.

Reply:
This might actually work since you have `toStartOfDay` in the sort key here the data is already duplicated so the data on disk for the column being Gorilla encoded might be a little bit of a win...but again. I prob wouldn't optimize things too much here for a 30 day TTLed table

Reply:
Gorilla is XOR based and the docs say "The smaller the difference between consecutive values is, i.e. the slower the values of the series changes, the better the compression rate" which makes me think it might make sense here (these values will change slowly but don't have a constant stride.)

T64 also looks interesting: https://clickhouse.com/docs/en/sql-reference/statements/create/table#t64

I'll leave this and the other columns from https://github.com/PostHog/posthog/pull/22163#discussion_r1593163578 alone for now — we can play with them later to see their impact.